src/storage-controller/src/lib.rs

```diff
@@ -193,9 +195,12 @@ pub struct Controller<T: Timestamp + Lattice + Codec64 + From<EpochMillis> + Tim
     // TODO(aljoscha): Should these live somewhere else?
     introspection_tokens: BTreeMap<GlobalId, Box<dyn Any + Send + Sync>>,
 
+    // The following two fields must always be locked in order.
```

Comment:
This feels spooky, is it possible we could pull both statistics into their own struct?

Reply:
Let me see if I can come up with something nice